doc/rtd/topics/configuration.rst

```diff
@@ -0,0 +1,81 @@
+.. _configuration:
+
+Configuration
+*************
+
+Internally, cloud-init builds a single configuration that is then referenced
+throughout the life of cloud-init. The configuration is built from multiple
+sources such that if a key is defined in multiple sources, the higher priority
+source overwrites the lower priority source.
+
+From lowest priority to highest, configuration sources are:
+
+* **Hardcoded config**: Config_ that lives within the source of cloud-init
+  and cannot be changed.
+* **Configuration directory**: Anything defined in ``/etc/cloud/cloud.cfg`` and
+  ``/etc/cloud/cloud.cfg.d``
+* **Runtime config**: Anything defined in ``/run/cloud-init/cloud.cfg``
+* **Kernel command line**: On the kernel command line, anything found between
+  ``cc:`` and ``end_cc`` will be interpreted as cloud-config user data.
+
+These four sources make up the **base configuration**. Added to these
+to provide the full configuration are:
+
+* **Vendor data**: :ref:`Data<vendordata>` provided by the datasource
+* **User data**: :ref:`Data<user_data_formats>` also provided by
+  the datasource
+
+.. note::
```

Comment:
I found this note a bit difficult to understand. 

As said in the previous paragraph, base configuration is merged from 4 sources, not only `/etc/cloud/cloud.cfg`. Reading this, I ask my self, have the other 3 sources to follow _the specific user data format_?

Maybe it's worth rephrasing this to something in the lines of:

base-configuration sources do not conform to :ref:`#cloud-config<topics/format:Cloud Config Data>`, ...

Reply:
Thanks. I didn't love the wording either, so you make a good point. The impetus for it is that before we had jinja templating in /etc/cloud*, multiple people have confused the #cloud-config mentioned [here](https://cloudinit.readthedocs.io/en/latest/topics/instancedata.html#using-instance-data) with /etc/cloud/cloud.cfg. Part of that confusion probably has something to do with them both being casually referred to as "cloud config".

I update that section as well.